src/lib/execPromise.ts

```diff
@@ -126,32 +140,39 @@ export const execPromise = ({
       }
 
       if (stderr.includes("linker `cc` not found")) {
-        rej(
-          new CLIError(
-            `\n${color.yellow(MARINE_CARGO_DEPENDENCY)} requires ${color.yellow(
-              "build-essential"
-            )} to be installed. Please install it and try again.\nOn debian-based systems (e.g. Ubuntu) you can install it using this command:\n\n${color.yellow(
-              "sudo apt install build-essential"
-            )}\n`
-          )
-        );
+        res({
+          expectedErrorMessage: `\n${color.yellow(
+            MARINE_CARGO_DEPENDENCY
+          )} requires ${color.yellow(
+            "build-essential"
+          )} to be installed. Please install it and try again.\nOn debian-based systems (e.g. Ubuntu) you can install it using this command:\n\n${color.yellow(
+            "sudo apt install build-essential"
+          )}\n`,
+        });
 
         return;
       }
 
       if (code !== 0) {
-        rej(
-          new Error(
-            `${printOutput ? "" : stderr}\nprocess exited${
-              code === null ? "" : ` with code ${code}`
-            }\n${debugInfo}`
-          )
-        );
-
-        return;
+        res({
+          errorMessage: `${getCommandFailedMessage(code)}${getErrorMessage(
+            printOutput,
+            stderr
+          )}`,
+        });
       }
 
-      res(stdout);
+      res({ success: stdout });
```

Comment:
maybe name it `stdout`?

later you have `return result.success`, and it's pretty hard to guess that `success` means `stdout`.